lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -160,6 +463,94 @@ export default function getAPIImplementation(): APIInterface {
                 userContext,
             });
 
+            if (response.status === "OK") {
```

Comment:
the token consumption API called on top will just consume the token and not actually change the password.. see diagram. So the API in the core will need to change to just consume the token and return OK or INVALID_TOKEN. 

Therefore, the recipe interface will also have to change to rename resetPasswordUsingToken to something like `consumePasswordResetToken`